book/src/specs/kimchi.md

```diff
@@ -1196,6 +1196,90 @@ Given value `v2` the layout looks like this
 where the notation `v2ci` and `v2pi` defined in the "Layout" section above.
 
 
+#### Foreign Field Addition
+
+These circuit gates are used to constrain that
+
+    $$left_input +/- right_input = field_overflow * foreign_modulus + result$$
+
+ Documentation:
+
+  For more details please see the [FFadd RFC](../rfcs/ffadd.md)
+
+  Mapping:
+    To make things clearer, the following mapping between the variable names
+    used in the code and those of the document can be helpful.
+
+```text
+    left_input_lo -> a0  right_input_lo -> b0  result_lo -> r0  bound_lo -> u0
+    left_input_mi -> a1  right_input_mi -> b1  result_mi -> r1  bound_mi -> u1
+    left_input_hi -> a2  right_input_hi -> b2  result_hi -> r2  bound_hi -> u2
+
+    field_overflow  -> x
+    sign            -> s
+    carry_lo        -> c0
+    carry_mi        -> c1
+    bound_carry_lo -> k0
+    bound_carry_mi -> k1
+```
+
+Let `left_input_lo`, `left_input_mi`, `left_input_hi` be 88-bit limbs of the left element
+
+Let `right_input_lo`, `right_input_mi`, `right_input_hi` be 88-bit limbs of the right element
+
+Let `foreign_modulus_lo`, `foreign_modulus_mi`, `foreign_modulus_hi` be 88-bit limbs of the foreign modulus
+
+Then the limbs of the result are
+
+- `result_lo = left_input_lo +/- right_input_lo - field_overflow * foreign_modulus_lo - 2^{88} * result_carry_lo`
+- `result_mi = left_input_mi +/- right_input_mi - field_overflow * foreign_modulus_mi - 2^{88} * result_carry_mi + result_carry_lo`
+- `result_hi = left_input_hi +/- right_input_hi - field_overflow * foreign_modulus_hi + result_carry_mi`
+
+`field_overflow` $=0$ or $1$ or $-1$ handles overflows in the field
+
+`carry_i` $= -1, 0, 1$ are auxiliary variables that handle carries between limbs
+
+Appart from the rangechecks of the chained inputs, we need to do an additional range check for a final bound
+to make sure that the result is less than the modulus, by adding `2^{3*88} - foreign_modulus` to it.
+ (This can be computed easily from the limbs of the modulus)
+Note that `2^{264}` as limbs reprsents: (0, 0, 0, 1) then:
+
+The upper-bound check can be calculated as
+- `bound_lo = result_lo - foreign_modulus_lo - bound_carry_lo * 2^{88}`
```

Comment:
Isn't this supposed to be 
`bound_log = result_lo + g_lo - bound_carry_lo * 2^{88}`

Reply:
yeah but g_lo is all zero